### ISSUE TITLE
allow mocked plugins (for `oc dev`) to run code

### ISSUE THREAD

**Issue by andyroyle** [opentable/oc#129]
Rather than just returning a static value, it would be nice to have the option to run a function.

**Comment by matteofigus** [comment]
:+1: 
current signature is:
```js
{
  "registries": ["http://registry.com"],
  "mocks": {
    "plugins": {
      "static": {
        "foo": "bar"
      }
    }
  }
}
```

I would suggest something that
* allows inclusion of js file
* perhaps a single file to rule them all (this means that for given project it would be possible to create and share that single file easily).
```js
{
  "registries": ["http://registry.com"],
  "mocks": {
    "plugins": {
      "src": "./mock-plugins.js",
      "static": {
        "bar": "hello"
      }
    }
  }
}
```
Where the requested file would be a node file with a oc-plugin-style-ish api for register method for each plugin name:
```js
module.exports.pluginFoo = [{
  name: 'foo',
  register: require('oc-foo'), // an actual plugin or a function with the mock
  options: { /* some options in case we use the real plugin */}
}, function(err){
  if(!!err){
    console.log('plugin failed to initialise');
    process.exit(1);
  }
}];
```

Another approach would be to have an `oc.js` instead of `oc.json` and directly extend the current api to accept a function. This sounds a little scary, first seems not bad and easier.

What do you think?

**Comment by andyroyle** [comment]
I would say we want to keep the plugins enumerated in the oc.json:

```json
{
  "registries": ["http://registry.com"],
  "mocks": {
    "plugins": {
      "dynamic": {
        "plugin1": "./plugin1.js",
        "plugin1": "./plugin2.js"
      },
      "static": {
        "bar": "hello"
      }
    }
  }
}
```

Do we need the register function for these plugins? Considering they're supposed to be mocks?

Can we keep the signature dead simple:

```javascript
module.exports = function(a){
  return a ? "foo" : "bar";
}
```

**Comment by matteofigus** [comment]
uhm, I actually like this, seems simple and readable :+1:

**Comment by andyroyle** [comment]
good because it took me 10 mins :+1: 

https://github.com/opentable/oc/compare/mock-plugins-code?expand=1

Just need to update the docs and I'll send a PR

**Comment by andyroyle** [comment]
closed by #130
